README.md
Pockeemuns
==========

Text adventure game emulating the world of pokemon.


Changelog
* Main.java file created with basic structure. : ERIC
* roughdraft.java file created with a GUI, improvements and organization : NIGEL
* Main.java renamed to roughsdraft.java and roughdraft.java renamed to Main.java : NIGEL


test.java
import java.awt.Desktop;
import java.net.URI;
import java.lang.Object;
import javax.swing.JOptionPane;
class BrowseURL
{
  public static void main(String args[]) throws Exception
 {
  String foo = JOptionPane.showInputDialog(null,"Enter a website you want to go to!", "Enter URL here!");
  Sasdfasdfasdf
 Desktop d=Desktop.getDesktop();

 if (foo.matches(".*http://.*")) {
    d.browse(new URI(foo));
 }
 else {
 d.browse(new URI("http://" + foo));
 }
 }
}


clicktest.java
//click mechanics by Nigel
import java.awt.FlowLayout;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.UIManager;


public class clicktest extends JFrame implements ActionListener{
	int clicked = 0;
	public void clicktest() {
		 String aString = Integer.toString(clicked);
		setTitle("Pokeemun attack mechanics");
		setSize(450,125);
		JLabel save = new JLabel();
		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
			FlowLayout flo = new FlowLayout();
		JButton button  = new JButton("ATTACK!");
		JLabel pageLabel = new JLabel(aString, (int)JLabel.CENTER_ALIGNMENT);
		button.addActionListener(new ActionListener() {
			@Override
			public void actionPerformed(ActionEvent arg0) {
			    clicked++;
			    String aString = Integer.toString(clicked);
			   pageLabel.setText(aString);
			    System.out.println(aString);
			}
		
			});
		
		setResizable( false );
		
		add(pageLabel);
		setLayout(flo);
		setVisible(true);
		add(save);
		add(button);
	}
	
	

	private void setLookAndFeel() {
		try {
			UIManager.setLookAndFeel(
					"com.sun.java.swing.plaf.nimbus.NimbusLookAndFeel"
					);
		}
		catch (Exception exc) {
		}
		
	
	}
	public static void main(String[] args) {
		clicktest click = new clicktest();
		click.clicktest();
	}



	@Override
	public void actionPerformed(ActionEvent arg0) {
		// TODO Auto-generated method stub
		
	}
}


roughdraft.java
package roughdraft;

import java.util.Scanner;

import javax.swing.JOptionPane;

public class roughdraft {
//introduction function
	public static void introduction() {
		int k =1;
		//START OF INTRODUCTION
		while(k!=0)
		{
		
		String name = JOptionPane.showInputDialog(null,"Welcome adventurer! Give me your name!","Access save code: 'paly'");
	    String confirmname = JOptionPane.showInputDialog("Your name is "+name+". Is that correct? 'Yes' or 'No'?");
	    
	     if(confirmname.equalsIgnoreCase("yes"))
	     {
	    	//START OF RIVAL NAME
	    	 String rivalname=JOptionPane.showInputDialog(null,"You have a friend that lives next to you! What is your friends name?");
	    	 JOptionPane.showMessageDialog(null,"Ah, I see! Your friends name is " + rivalname+"! Your new adventure awaits, " + name+"!");
	    	 k=0;
	     }
	     }
		//END OF INTRODUCTION
		//START OF CHOOSING POKEMEN
		while(k==0)
		{
		String r = JOptionPane.showInputDialog("You make your way to Prof. Oak's office. \n What pokemon do you want? 'Charmander' , 'Bulbasaur' , or 'Squirtle'?");
		
	    if(r.equalsIgnoreCase("charmander"))
	    {

	    }
	    else if(r.equalsIgnoreCase("bulbasaur"))
	    {

	    }
	    else if(r.equalsIgnoreCase("squirtle"))
	    {

	   	 	String confirmsq = JOptionPane.showInputDialog("You have chosen " +r+", are you sure? 'Yes' or 'no'?");
	   	 	if (confirmsq.equalsIgnoreCase("no")) {
	   	 		k=0;
	   	 	}
	   	 	else if(confirmsq.equalsIgnoreCase("yes")){
	   	 	JOptionPane.showMessageDialog(null, "You and your " + r +" will now spend the rest of your far journey together!");
	   	 		k=1;
	   	 	}
	   	 	else {
	   	 		k=0;
	   	 	}
	    }
	    else{
	    JOptionPane.showMessageDialog(null,"Sorry, I didn't get that.");
		k=0;
	    }
		}
		//END OF CHOOSING POKEMEN.
	}
// main function
	public static void main(String[] args) {
		String title= JOptionPane.showInputDialog("Welcome to the game! \n If you want to load a save, enter your access checkpoint!\n Otherwise, you can type in anything to start a new game.");
		switch (title) {
		case "paly": 
			rd test = new rd();
			rd.introduction();
			break;
		default:
			JOptionPane.showInputDialog("Welcome to the game of- POKEMANS!");
		}
		rd test = new rd();
		rd.introduction();
		}
	}



roughsdraft2.java
package roughdraft;

import java.util.Scanner;

import javax.swing.JOptionPane;

public class roughdraft {
//introduction function
	public static void introduction() {
		int k =1;
		//START OF INTRODUCTION
		while(k!=0)
		{
		
		String name = JOptionPane.showInputDialog(null,"Welcome adventurer! Give me your name!","Access save code: 'paly'");
	    String confirmname = JOptionPane.showInputDialog("Your name is "+name+". Is that correct? 'Yes' or 'No'?");
	    
	     if(confirmname.equalsIgnoreCase("yes"))
	     {
	    	//START OF RIVAL NAME
	    	 String rivalname=JOptionPane.showInputDialog(null,"You have a friend that lives next to you! What is your friends name?");
	    	 JOptionPane.showMessageDialog(null,"Ah, I see! Your friends name is " + rivalname+"! Your new adventure awaits, " + name+"!");
	    	 k=0;
	     }
	     }
		//END OF INTRODUCTION
		//START OF CHOOSING POKEMEN
		while(k==0)
		{
		String r = JOptionPane.showInputDialog("You make your way to Prof. Oak's office. \n What pokemon do you want? 'Charmander' , 'Bulbasaur' , or 'Squirtle'?");
		
	    if(r.equalsIgnoreCase("charmander"))
	    {

	    }
	    else if(r.equalsIgnoreCase("bulbasaur"))
	    {

	    }
	    else if(r.equalsIgnoreCase("squirtle"))
	    {

	   	 	String confirmsq = JOptionPane.showInputDialog("You have chosen " +r+", are you sure? 'Yes' or 'no'?");
	   	 	if (confirmsq.equalsIgnoreCase("no")) {
	   	 		k=0;
	   	 	}
	   	 	else if(confirmsq.equalsIgnoreCase("yes")){
	   	 	JOptionPane.showMessageDialog(null, "You and your " + r +" will now spend the rest of your far journey together!");
	   	 		k=1;
	   	 	}
	   	 	else {
	   	 		k=0;
	   	 	}
	    }
	    else{
	    JOptionPane.showMessageDialog(null,"Sorry, I didn't get that.");
		k=0;
	    }
		}
		//END OF CHOOSING POKEMEN.
	}
// main function
	public static void main(String[] args) {
		String title= JOptionPane.showInputDialog("Welcome to the game! \n If you want to load a save, enter your access checkpoint!\n Otherwise, you can type in anything to start a new game.");
		switch (title) {
		case "paly": 
			rd test = new rd();
			rd.introduction();
			break;
		default:
			JOptionPane.showInputDialog("Welcome to the game of- POKEMANS!");
		}
		rd test = new rd();
		rd.introduction();
		}
	}


program/GUIprogram.java
//concept GUI http://i.imgur.com/d37J5x6.png 
// proj by Nigel
import java.awt.Desktop;
import java.awt.FlowLayout;
import java.awt.Frame;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.IOException;
import java.net.URISyntaxException;
import java.net.URL;

import javax.swing.*;

public class GUIprogram extends JFrame {
	

	public GUIprogram() {

		
		setTitle("Pokeemans");
		setSize(450,125);
		JLabel pageLabel = new JLabel("Pokeemuns game 0.1 ALPHA", (int) JLabel.CENTER_ALIGNMENT);
		JLabel save = new JLabel("Enter your checkpoint access code if you have one.");
		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		JTextField checkpoint = new JTextField(10);
		FlowLayout flo = new FlowLayout();
		setResizable( false );
		setLayout(flo);
		JButton checkpoints = new JButton("Play!");
		JButton test = new JButton("New Game!");
		test.addActionListener(new ActionListener() {
			@Override
			public void actionPerformed(ActionEvent arg0) {
				
				roughdraft testlol = new roughdraft();
				testlol.introduction();
				System.exit(0);
			}

		
				
			});
			

		JButton test1 = new JButton("GitHub");
		test1.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e)
            {
            	try {
					Desktop.getDesktop().browse(new URL("https://github.com/Compurnee/Pockeemuns").toURI());
				} catch (IOException | URISyntaxException e1) {
					e1.printStackTrace();
				}
            	System.exit(0);
            }
        });
		add(pageLabel);
		add(test);
		add(test1);
		add(save);
		add(checkpoint);
		add(checkpoints);
		
		setVisible(true);
	}
	

	private void setLookAndFeel() {
		try {
			UIManager.setLookAndFeel(
					"com.sun.java.swing.plaf.nimbus.NimbusLookAndFeel"
					);
		}
		catch (Exception exc) {
			//ignore error
		}
		
	}
	public static void main(String[] args) {
		GUIprogram frame = new GUIprogram();
	}
}


program/Main.java
//concept GUI http://i.imgur.com/d37J5x6.png 
// proj by Nigel
import java.awt.Desktop;
import java.awt.FlowLayout;
import java.awt.Frame;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.IOException;
import java.net.URISyntaxException;
import java.net.URL;

import javax.swing.*;

public class GUIprogram extends JFrame {
	

	public GUIprogram() {

		
		setTitle("Pokeemans");
		setSize(450,125);
		JLabel pageLabel = new JLabel("Pokeemuns game 0.1 ALPHA", (int) JLabel.CENTER_ALIGNMENT);
		JLabel save = new JLabel("Enter your checkpoint access code if you have one.");
		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		JTextField checkpoint = new JTextField(10);
		FlowLayout flo = new FlowLayout();
		setResizable( false );
		setLayout(flo);
		JButton checkpoints = new JButton("Play!");
		checkpoints.addActionListener(new ActionListener() {
			@Override
			public void actionPerformed(ActionEvent arg0) {
				
				JOptionPane.showMessageDialog(null, "Button not working");
			
			}
		});
		JButton test = new JButton("New Game!");
		test.addActionListener(new ActionListener() {
			@Override
			public void actionPerformed(ActionEvent arg0) {
				
				roughdraft testlol = new roughdraft();
				testlol.introduction();
				System.exit(0);
			}

		
				
			});
			

		JButton test1 = new JButton("GitHub");
		test1.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e)
            {
            	try {
					Desktop.getDesktop().browse(new URL("https://github.com/Compurnee/Pockeemuns").toURI());
				} catch (IOException | URISyntaxException e1) {
					e1.printStackTrace();
				}
            	
            }
        });
		JButton exit = new JButton("Exit");
		exit.addActionListener(new ActionListener() {
			@Override
			public void actionPerformed(ActionEvent arg0) {
				
				System.exit(0);
			}

		
				
			});
			
		add(pageLabel);
		add(test);
		add(test1);
		add(save);
		add(checkpoint);
		add(checkpoints);
		add(exit);
		
		setVisible(true);
	}
	

	private void setLookAndFeel() {
		try {
			UIManager.setLookAndFeel(
					"com.sun.java.swing.plaf.nimbus.NimbusLookAndFeel"
					);
		}
		catch (Exception exc) {
			//ignore error
		}
		
	}
	public static void main(String[] args) {
		GUIprogram frame = new GUIprogram();
	}
}
